From ebeff627c5df410ad8ec872e530125472c5152b7 Mon Sep 17 00:00:00 2001
From: Alejandro Sierra <b18039@freescale.com>
Date: Tue, 19 Jan 2016 17:58:22 -0600
Subject: [PATCH 23/37] clk-imx6: ldb clock root from mmdc1_axi_clk_root

clk-imx6: ldb clock root from mmdc1_axi_clk_root.
Performance improvement on IPU and GPU clock configurations.

Signed-off-by: Alejandro Sierra <b18039@freescale.com>
---
 arch/arm/mach-imx/clk-imx6q.c | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index bc3feec..6b89804 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -808,8 +808,19 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 		imx_clk_set_parent(clk[IMX6QDL_CLK_IPU2_SEL], clk[IMX6QDL_CLK_PLL2_PFD2_396M]);
 		imx_clk_set_rate(clk[IMX6QDL_CLK_IPU2], 200000000);
 	} else if (cpu_is_imx6q()) {
-		imx_clk_set_parent(clk[IMX6QDL_CLK_IPU1_SEL], clk[IMX6QDL_CLK_MMDC_CH0_AXI]);
-		imx_clk_set_parent(clk[IMX6QDL_CLK_IPU2_SEL], clk[IMX6QDL_CLK_MMDC_CH0_AXI]);
+                imx_clk_set_rate(clk[IMX6QDL_CLK_PLL3_PFD1_540M], 540000000);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_IPU1_SEL], clk[IMX6QDL_CLK_PLL3_PFD1_540M]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_IPU2_SEL], clk[IMX6QDL_CLK_PLL3_PFD1_540M]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_AXI_ALT_SEL], clk[IMX6QDL_CLK_PLL3_PFD1_540M]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_AXI_SEL], clk[IMX6QDL_CLK_AXI_ALT_SEL]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_LDB_DI0_DIV_SEL], clk[IMX6QDL_CLK_LDB_DI0_DIV_3_5]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_LDB_DI1_DIV_SEL], clk[IMX6QDL_CLK_LDB_DI1_DIV_3_5]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_IPU1_DI0_SEL], clk[IMX6QDL_CLK_LDB_DI0]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_IPU1_DI1_SEL], clk[IMX6QDL_CLK_LDB_DI1]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_IPU2_DI0_SEL], clk[IMX6QDL_CLK_LDB_DI0]);
+                imx_clk_set_parent(clk[IMX6QDL_CLK_IPU2_DI1_SEL], clk[IMX6QDL_CLK_LDB_DI1]);
+                /* set eim_slow to 132Mhz */
+                imx_clk_set_rate(clk[IMX6QDL_CLK_EIM_SLOW], 132000000);
 	}
 
 	imx_clk_set_parent(clk[IMX6QDL_CLK_AXI_ALT_SEL], clk[IMX6QDL_CLK_PLL3_PFD1_540M]);
@@ -850,8 +861,8 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 		} else {
 			imx_clk_set_parent(clk[IMX6QDL_CLK_GPU3D_SHADER_SEL], clk[IMX6QDL_CLK_PLL2_PFD1_594M]);
 			imx_clk_set_rate(clk[IMX6QDL_CLK_GPU3D_SHADER], 594000000);
-			imx_clk_set_parent(clk[IMX6QDL_CLK_GPU3D_CORE_SEL], clk[IMX6QDL_CLK_MMDC_CH0_AXI]);
-			imx_clk_set_rate(clk[IMX6QDL_CLK_GPU3D_CORE], 528000000);
+			imx_clk_set_parent(clk[IMX6QDL_CLK_GPU3D_CORE_SEL], clk[IMX6QDL_CLK_PLL3_USB_OTG]);
+                        imx_clk_set_rate(clk[IMX6QDL_CLK_GPU3D_CORE], 480000000);
 			imx_clk_set_parent(clk[IMX6QDL_CLK_GPU2D_CORE_SEL], clk[IMX6QDL_CLK_PLL3_USB_OTG]);
 			imx_clk_set_rate(clk[IMX6QDL_CLK_GPU2D_CORE], 480000000);
 		}
-- 
1.9.1

